geonode/tests/utils.py

```diff
@@ -227,6 +227,19 @@ def upload_file(self, _file):
             data=params,
             ajax=True,
             force_login=True)
+
+        #closes the files
+        if ext.lower() == '.shp':
+            for spatial_file in spatial_files:
+                ext, _ = spatial_file.split('_')
```

Comment:
Seems unnecessary to compute the filename if `params[spatial_file]` and `params['tif_file']` can be checked to see if they are open file descriptors. Something along the lines of
```python
for spatial_file in spatial_files:
    if isinstance(params[spatial_file], file):
        params[spatial_files].close()

if isinstance(params['tif_file'], file):
    params['tif_file'].close()